### ISSUE TITLE
Add support for three way merge

### ISSUE THREAD

**Issue by ackvf** [eamodio/vscode-gitlens#914]
It's quite difficult to do a merge between _current_ and _incoming_ without the knowledge of the common ancestor's state. It would be nice if git lens helped me find and display these compares more easily:

```
Base <- Current
Base <- Incoming
Current <- Incoming
```

Thank you

**Comment by ackvf** [comment]
Turns out it was just my lack of knowledge and it is actually a matter of git config.

`$ git config --global merge.conflictstyle diff3`

How?
https://stackoverflow.com/questions/18131515/how-can-i-see-a-three-way-diff-for-a-git-merge-conflict

Why?
https://stackoverflow.com/questions/4129049/why-is-a-3-way-merge-advantageous-over-a-2-way-merge


---

Anyway, if there was a first-class support from within the extension, to be able to see these views more efficiently, it would still be better. The git config makes it all inline in one file, I would still prefer to see it in separate tabs.

**Comment by ishamf** [comment]
Yes, better support will be nice! I think just having 3 buttons (Base - Current, Base - Incoming, Current - Incoming) to compare if it's a 3-way merge would be very helpful.

**Comment by eamodio** [comment]
This isn't something that I will likely add myself, but if the community wants to provide a PR I'm certainly open to it -- depending on the complexity and scope of the changes required.
